app/src/lib/git/status.ts

```diff
@@ -75,9 +67,28 @@ export async function getStatus(repository: Repository): Promise<IStatusResult>
   for (const entry of parsePorcelainStatus(result.stdout)) {
     if (entry.kind === 'entry') {
       const status = mapStatus(entry.statusCode)
+
+      if (status.kind === 'ordinary') {
+        if (status.staged === GitStatusEntry.Added
+            && status.unstaged === GitStatusEntry.Deleted) {
+          // we can safely avoid drawing this file in the app
```

Comment:
It depends on whether the content of the untracked file with the same path has changed:

 - no content change - no-op
 - content change - modified

I'd need to look at what diff we render here at this stage. It could be confusing.